### ISSUE TITLE
No Console on Beta

### ISSUE THREAD

**Issue by aes512** [opnsense/core#1310]
I just tried to boot the serial version of the new beta on my Netgate RCC-VE-2440 (USB serial only). It doesn't provide a serial console. 

Are we just SOL with these Netgate/ADI devices? (don't bother hinting at opnsense-bootstrap'ing, that doesn't work with these devices either).

**Comment by fichtner** [comment]
This isn't enough information. Did it work with 16.7? 16.1? Is this your first try?

In general, we provide a vanilla FreeBSD environment. If it boots there and everything works as expected, the chances are good OPNsense will too.

Did you read this?

https://www.netgate.com/docs/rcc-ve-2440/opnsense.html

If yes, what doesn't work in particular?


Cheers,
Franco

**Comment by aes512** [comment]
Yeah, I read it. They lost me at 15.7.18 and i386. It hasn't worked on any version, there was talk on IRC that it was going to be supported when this beta was initially released, but it looks like thatis not true.

**Comment by fichtner** [comment]
ask the, to update their guide :)

**Comment by aes512** [comment]
Or not. I was hopeful that there would be broader support, but we'll likely toss this device in the garbage and pick a different hardware platform, and stick with rolling OpenBSD routers for the project that we were looking to use OPNsense for. Thanks anyways though.

**Comment by fichtner** [comment]
I asked for details in questions, I still don't have them so I don't know how to help other than show compassion...

https://forum.opnsense.org/index.php?topic=3198.msg10393#msg10393

From what I can tell, /boot/loader.conf.local only requires the following:

```
comconsole_port="0x2F8"
hint.uart.0.flags="0x00"
hint.uart.1.flags="0x10"
```

It would be of help to tell which part doesn't work: install, first boot, subsequent boots, etc.

**Comment by aes512** [comment]
Console doesn't work - i.e. - dd the serial img to USB > boot the device from USB > console is output is lost once the device is booted from the USB stick. There is no way to manage / interact with the system after it boots into OPNsense.

I don't see any way to add those changes, unless you are implying that users would have to unpack the image, add those to the /boot/loader.conf.local file, and the repack the image? Otherwise, I do not see any other method to get those changes on the USB stick.

**Comment by fichtner** [comment]
Do you see the boot loader splash screen with the menu?

**Comment by aes512** [comment]
Again, no I do not. I see the bios > (hit F12 for the boot menu) > choose USB boot > then nothing

**Comment by aes512** [comment]
FWIW the USB stick works in terms of it's ability to boot into the OS, console does now work.

**Comment by fichtner** [comment]
Sorry for asking seemingly spurious questions... serial images work with multiple devices, I just need to understand if this is new and where the issue is located.

Did this work for 16.7 serial images? Are you sure this is the serial image?

If both of these are yes we are looking at an issue with FreeBSD 11.0 here we haven't seen before.

I will try my best to offer support for all ranges of devices. It's just that I don't have this particular device to test and see and fix, so please understand my situation and surprise. I'm at your mercy to get to the bottom of this and iron out this flaw in the beta images so that we can deliver something that works reliable in this case for the final release.


Cheers,
Franco

**Comment by aes512** [comment]
This did not work for any of the OPNsense serial images to date on my Netgate router. The image(s) do work on another box that I have that has a serial out, but no luck with the Netgate/ADI device.

**Comment by fichtner** [comment]
This is a single character twist on the 17.1-BETA serial image for amd64, can you try it?

https://pkg.opnsense.org/snapshots/OPNsense-201612182349-OpenSSL-serial-amd64.img.bz2

**Comment by aes512** [comment]
@fichtner sure, I'll give it a spin later today and let you know the results. Thank you!

**Comment by aes512** [comment]
@fichtner that image did not work either. No console ever shows up. 

Here is a question - in the beta release notes, it is stated that SSH is available for headless installation. When the USB serial img boots, does it start networking automatically? I tried to arping ff:ff:ff:ff:ff:ff when I booted the beta, but I never got any signs of networking from my router. Would there a way to get SSH working automatically? Otherwise, this really sucks. 

There does not seem to be any way to get OPNsense installed directly on the appliances, as I would have to install it on another machine to the Netgate's SSD :(

**Comment by fichtner** [comment]
I'm not sure what they do to their images to properly boot their hardware. If their guide doesn't work anymore, now is the time to ask them for clarification or look for alternatives that run smoothly on that box.

If you have two interfaces, the one that gives you a 192.168.1.0/24 DHCP lease is the port that has the SSH installer, it runs on 192.168.1.1.

Without looking at the console it's difficult to say if SSH (or the system itself) came up ok or bails while trying to set up something else for the hardware.

**Comment by fichtner** [comment]
installer user / password is:

installer
opnsense

**Comment by lattera** [comment]
Hey @aes512, I've got a few Netgate appliances that I can test on. It might take me a day or two to find them, though.

**Comment by aes512** [comment]
@fichtner pfSense has "serial" images specific for Netgate/ADI, but I think that the only differences are the loader.conf and loader.conf.local files that they ship with:

````
$ cat /boot/loader.conf
autoboot_delay="3"
boot_serial="YES"
console="comconsole"
comconsole_speed="115200"
comconsole_port="0x2F8"
hint.uart.0.flags="0x00"
hint.uart.1.flags="0x10"
hw.usb.no_pf="1"

$ cat /boot/loader.conf.local
kern.cam.boot_delay=10000
kern.ipc.nmbclusters="1000000"
````

I'm fairly certain that comconsole_speed="115200" and comconsole_port="0x2f8" are all that would be needed to boot. That said, I'm wondering if I can mount the OPNsense USB stick rw to pfSense and make those changes to the loader configs?

**Comment by djGrrr** [comment]
You should be able to mount the OPNsense USB stick on pfSense or any FreeBSD system and modify the loader.conf file to test this out

**Comment by aes512** [comment]
I was able to mount the USB stick last night and add the customer loader.conf.local file that was needed. This allowed me to obtain console access to the router finally.

Unfortunately, this was all a huge waste of time. OPNsense would not pass any traffic on my igb interfaces that were part of a lagg group in VLANs! Separate bug, this works just fine with pfSense. I also triggered some kind of kernel panic boot loops when I changed the initial subnet that I setup for basic (initial) LAN connectivity. All of this makes me too nervous to continue with this distro, I don't have time to beta test in prod :(

Thanks for the help
